BootloaderCorePkg/Stage2/Stage2Support.c

```diff
@@ -607,16 +609,12 @@ BuildExtraInfoHob (
     }
   }
 
-  // Build Performance Hob
-  Count  = LdrGlobal->PerfData.PerfIndex;
-  Length = sizeof (PERFORMANCE_INFO) + sizeof (UINT64) * Count;
-  PerformanceInfo = BuildGuidHob (&gLoaderPerformanceInfoGuid, Length);
-  if (PerformanceInfo != NULL) {
-    PerformanceInfo->Revision = 1;
-    PerformanceInfo->Count = (UINT16)LdrGlobal->PerfData.PerfIndex;
-    PerformanceInfo->Frequency = LdrGlobal->PerfData.FreqKhz;
-    PerformanceInfo->Flags = 0;
-    CopyMem (PerformanceInfo->TimeStamp, LdrGlobal->PerfData.TimeStamp, sizeof (UINT64) * Count);
+  // Build device table Hob
+  DeviceTable  = (SBL_DEVICE_TABLE *)LdrGlobal->DeviceTable;
+  Length = sizeof (SBL_DEVICE_TABLE) + sizeof (SBL_DEVICE) * DeviceTable->DeviceNumber;
```

Comment:
Do we need to create a hob if DeviceNumber is 0?

Reply:
yes. we should create HOB even device number is 0 since payload will get this hob, and could know the device number is 0 from in the table header.